core/debug/src/debug.c

```diff
@@ -421,3 +421,44 @@ void debug_fault(THaltError reason, uint32_t lr, uint32_t sp)
 #endif /* defined(ARCH_MPU_ARMv8M) */
     DEBUG_PRINT_END();
 }
+
+bool debug_collect_halt_info(uint32_t lr, uint32_t sp, THaltInfo *halt_info)
+{
+    /* For security reasons don't provide any information about faults in uVisor itself. */
+#if defined(ARCH_MPU_ARMv8M)
+    bool no_halt_info = EXC_FROM_S(lr);
+#else
+    bool no_halt_info = !EXC_FROM_PSP(lr);
+#endif /* defined(ARCH_MPU_ARMv8M) */
+
+    halt_info->valid_data = 0;
+    if (no_halt_info) {
+        return false;
```

Comment:
Note to self: We don't want to print uVisor faults, since it could expose secrets. To debug uVisor faults users will have to use a uVisor debug build with semihosting.